ignite/handlers/fbresearch_logger.py

```diff
@@ -0,0 +1,185 @@
+"""FBResearch logger and its helper handlers."""
+
+import datetime
+from typing import Any, Optional
+
+# from typing import Any, Dict, Optional, Union
+
+import torch
+
+from ignite.engine import Engine, Events
+from ignite.handlers import Timer
+
+
+MB = 1024.0 * 1024.0
+
+
+class FBResearchLogger:
+    """Logs training and validation metrics for research purposes.
+
+    This logger is designed to attach to an Ignite Engine and log various metrics
+    and system stats at configurable intervals, including learning rates, iteration
+    times, and GPU memory usage.
+
+    Args:
+        logger (logging.Logger): The logger to use for output.
+        delimiter (str): The delimiter to use between metrics in the log output.
+        show_output (bool): Flag to enable logging of the output from the engine's process function.
```

Comment:
We are using some sphinx plug-in such that we do not need to provide types in the docstring:
```suggestion
        logger: The logger to use for output.
        delimiter: The delimiter to use between metrics in the log output.
        show_output: Flag to enable logging of the output from the engine's process function.
```